server/ast/alter_table.go

```diff
@@ -37,3 +52,100 @@ func nodeAlterTableSetSchema(node *tree.AlterTableSetSchema) (vitess.Statement,
 	}
 	return nil, fmt.Errorf("ALTER TABLE SET SCHEMA is not yet supported")
 }
+
+// nodeAlterTableCmds converts tree.AlterTableCmds into a slice of vitess.DDL
+// instances that can be executed by GMS. |tableName| identifies the table
+// being altered, and |ifExists| indicates whether the IF EXISTS clause was
+// specified.
+func nodeAlterTableCmds(
+	node tree.AlterTableCmds,
+	tableName vitess.TableName,
+	ifExists bool) ([]*vitess.DDL, error) {
+
+	if len(node) == 0 {
+		return nil, fmt.Errorf("no commands specified for ALTER TABLE statement")
+	} else if len(node) > 1 {
+		return nil, fmt.Errorf("ALTER TABLE with multiple commands is not yet supported")
+	}
+
+	vitessDdlCmds := make([]*vitess.DDL, 0, len(node))
+	for _, cmd := range node {
+		switch cmd := cmd.(type) {
+		case *tree.AlterTableAddConstraint:
+			statement, err := nodeAlterTableAddConstraint(cmd, tableName, ifExists)
+			if err != nil {
+				return nil, err
+			}
+			vitessDdlCmds = append(vitessDdlCmds, statement)
+
+		default:
+			return nil, fmt.Errorf("ALTER TABLE with unsupported command type %T", cmd)
+		}
+	}
+
+	return vitessDdlCmds, nil
+}
+
+// nodeAlterTableAddConstraint converts a tree.AlterTableAddConstraint instance
+// into a vitess.DDL instance that can be executed by GMS. |tableName| identifies
+// the table being altered, and |ifExists| indicates whether the IF EXISTS clause
+// was specified.
+func nodeAlterTableAddConstraint(
+	node *tree.AlterTableAddConstraint,
+	tableName vitess.TableName,
+	ifExists bool) (*vitess.DDL, error) {
+
+	if node.ValidationBehavior == tree.ValidationSkip {
+		return nil, fmt.Errorf("NOT VALID is not supported yet")
+	}
+
+	switch constraintDef := node.ConstraintDef.(type) {
+	case *tree.UniqueConstraintTableDef:
+		return nodeUniqueConstraintTableDef(constraintDef, tableName, ifExists)
+	default:
+		return nil, fmt.Errorf("ALTER TABLE with unsupported constraint "+
+			"definition type %T", node)
+	}
+}
+
+// nodeUniqueConstraintTableDef converts a tree.UniqueConstraintTableDef instance
+// into a vitess.DDL instance that can be executed by GMS. |tableName| identifies
+// the table being altered, and |ifExists| indicates whether the IF EXISTS clause
+// was specified.
+func nodeUniqueConstraintTableDef(
```

Comment:
This seems unique enough to have its own file.

Reply:
Fixed in a2b63abb73671ada155ae3dc4d18c2d50f83f23a